float32 obs_min_dist
float32 obs_width